### PR TITLE
Add amdirt recipe

### PR DESCRIPTION
**Pull request by maxibor**
Adding recipe for AMDirT https://github.com/SPAAM-community/AMDirT
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Review by jfy133 — CHANGES_REQUESTED** [review]
Tests are failing but it would also be nice to get some pinned versions on there - we've already had issues with these during testing

**Comment by jfy133** [comment]
https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=32431&view=logs&j=e14e69ff-a0ae-55c4-b71d-229b239cfb2f&t=4dddc55b-10bc-50fc-ac68-7ff899b7a030&l=1088

@maxibor possibly a streamlit version error: 

```
15:06:25 BIOCONDA INFO (OUT) AttributeError: module 'streamlit' has no attribute 'cache_data'
```

By defaut it resolved to:

```
15:05:42 BIOCONDA INFO (OUT)     streamlit:                     1.8.0-pyhd8ed1ab_0          conda-forge
15:05:42 BIOCONDA INFO (OUT)     streamlit-aggrid:              0.3.4.post3-pyhd8ed1ab_0    conda-forge
```

which seems a bit old, no?

File: recipes/amdirt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  run:
+    - click
+    - colorlog
+    - jsonschema
+    - numpy
+    - pandas
+    - python >=3.9
+    - requests
+    - rich
+    - streamlit
```
Comment (jfy133) [inline recipes/amdirt/meta.yaml:32]:
```suggestion
    - streamlit >=1.21.0
```

**Comment by jfy133** [comment]
Tried setting to 1.21 as minimum streamlit version (But remaining dependencies should be reviewed)

**Comment by maxibor** [comment]
Installing it with pip will pick up the `package_data` from `setup.py`, but somehow won't be picked up using `conda build`...

With conda-build on azure:
```bash
[...]
Traceback (most recent call last):
08:29:04 BIOCONDA INFO (OUT)   File "/opt/conda/conda-bld/amdirt_1682756666019/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/bin/AMDirT", line 6, in <module>
08:29:04 BIOCONDA INFO (OUT)     from AMDirT.cli import cli
08:29:04 BIOCONDA INFO (OUT)   File "/opt/conda/conda-bld/amdirt_1682756666019/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.11/site-packages/AMDirT/cli.py", line 186, in <module>
08:29:04 BIOCONDA INFO (OUT)     type=click.Choice(get_table_list()),
08:29:04 BIOCONDA INFO (OUT)                       ^^^^^^^^^^^^^^^^
08:29:04 BIOCONDA INFO (OUT)   File "/opt/conda/conda-bld/amdirt_1682756666019/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.11/site-packages/AMDirT/cli.py", line 15, in get_table_list
08:29:04 BIOCONDA INFO (OUT)     with open(json_path, "r") as f:
08:29:04 BIOCONDA INFO (OUT)          ^^^^^^^^^^^^^^^^^^^^
08:29:04 BIOCONDA INFO (OUT) FileNotFoundError: [Errno 2] No such file or directory: '/opt/conda/conda-bld/amdirt_1682756666019/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.11/site-packages/AMDirT/assets/tables.json'
08:29:05 BIOCONDA INFO (OUT) Tests failed for amdirt-1.4-pyhdfd78af_0.tar.bz2 - moving package to /opt/conda/conda-bld/broken
[...]
```

Testing it on my local machine (no `tables.json` neither)
```bash
$ ls /Users/maxime/mambaforge/envs/bioconda/conda-bld/amdirt_1683028811095/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.10/site-packages/AMDirT/assets
__init__.py __pycache__ #no tables.json
```

Installing it with pip

```bash
$ which amdirt
/Users/maxime/mambaforge/envs/amdirt_pip/bin/amdirt
(amdirt_pip)  maxime@eva  ~  ls /Users/maxime/mambaforge/envs/amdirt_pip/lib/python3.11/site-packages/AMDirT/assets/
__init__.py __pycache__ tables.json #tables.json has been packaged as expected since defined in setup.py as package_data
```

**Comment by maxibor** [comment]
For future reference to anyone who ends reading this:
Pick up your poison, either use `include_package_data=True`, or `package_data`, but not both.

https://setuptools.pypa.io/en/latest/userguide/datafiles.html

**Review by jfy133 — APPROVED** [review]

**Pull request merged**